### ISSUE TITLE
Delete multiple runs function (`delete_runs()`)

### ISSUE THREAD

**Issue by natannvw** [mlflow/mlflow#10866]
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

There is only `mlflow.client.delete_run()`. I created a function `delete_runs()` that gets a list of run IDs and calls the `mlflow.client.delete_run()` with parallel executing:

```
def delete_run(run, mlflow_client: MlflowClient):
    mlflow_client.delete_run(run.info.run_id)


def delete_runs(runs, mlflow_client: MlflowClient):
    with ThreadPoolExecutor() as executor:
        list(
            tqdm(
                executor.map(
                    lambda run: delete_run(run, mlflow_client=mlflow_client), runs
                ),
                total=len(runs),
                desc="Deleting runs",
            )
        )
```

### Motivation

> #### What is the use case for this feature?
To efficiently delete multiple runs in an experiment.

> #### Why is this use case valuable to support for MLflow users in general?
It's useful and it's more efficient than calling the `mlflow.client.delete_run()` in regular loop.

> #### Why is this use case valuable to support for your project(s) or organization?
I delete multiple (100s and 1000s) runs that were failed or killed.

> #### Why is it currently difficult to achieve this use case?
It is not but should be in mlflow support for my opinion.

### Details

```
def delete_run(run, mlflow_client: MlflowClient):
    mlflow_client.delete_run(run.info.run_id)


def delete_runs(runs, mlflow_client: MlflowClient):
    with ThreadPoolExecutor() as executor:
        list(
            tqdm(
                executor.map(
                    lambda run: delete_run(run, mlflow_client=mlflow_client), runs
                ),
                total=len(runs),
                desc="Deleting runs",
            )
        )
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [X] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [X] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [X] `area/windows`: Windows support

### What language(s) does this bug affect?

- [X] `language/r`: R APIs and clients
- [X] `language/java`: Java APIs and clients
- [X] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [X] `integrations/azure`: Azure and Azure ML integrations
- [X] `integrations/sagemaker`: SageMaker integrations
- [X] `integrations/databricks`: Databricks integrations

**Comment by WeichenXu123** [comment]
Thanks, we can add this API, but I think your implementation is suboptimal,

why not make this a "batched operation" ? We can send batched run_ids to backend for executing deletion.

**Comment by natannvw** [comment]
Sounds good to me, but I'm not familiar with it. Can you refer me to any doc? I will try to implement it again.

**Comment by WeichenXu123** [comment]
@natannvw 

You can read `delete_run` code under `mlflow/store/tracking` directory,

For rest_store and sqlalchemy_store, for `delete_runs`, it should be able to pack multiple run_id into a list, 
For rest_store,
it should send the list to rest server backend,

For sqlalchemy_store , you should generate one sql that contains predicate like `delete ... where run_id in (run_id1, run_id2, ...)`

**Comment by WeichenXu123** [comment]
@natannvw 

In databricks , we have a API: https://docs.databricks.com/api/workspace/experiments/deleteruns

it deletes all runs that were **created prior to or at the specified timestamp**.

does it satisfy your use-case ? If yes we should implement this instead of your proposed API (delete specified run_id set)

**Comment by natannvw** [comment]
@WeichenXu123
No, because my goal is to scale up the `delete_run()` which is based on the run ID.

**Comment by WeichenXu123** [comment]
@natannvw Got it.

Then I suggest we just add a client side helper API like your original proposal, so that we don't need to change backend server code.

**Comment by natannvw** [comment]
@WeichenXu123 
Okay. What is my next step then? Copy my helper as is to the file and PR?

**Comment by WeichenXu123** [comment]
@natannvw Feel free to file PR according to suggestion in https://github.com/mlflow/mlflow/issues/10866#issuecomment-1903442418 and https://github.com/mlflow/mlflow/issues/10866#issuecomment-1903690994 

No hesitate to ask me if you have any blockers.

**Comment by natannvw** [comment]
@WeichenXu123 
Thanks, indeed I have some. I am not familiar (yet) with rest and how to implement my code exactly into mlflow.

**Comment by WeichenXu123** [comment]
@natannvw 

You can read code under `mlflow/store/tracking`, it needs to support file-store/rest-store/sqlalchemy-store .   You should firstly understand how `delete_run` is implemented for the 3 kinds of store backends.

**Comment by natannvw** [comment]
@WeichenXu123 
Hey,
I created a PR: Delete runs #11519
